### PR TITLE
Introduce maven central mirror parameter

### PR DESCRIPTION
**Pull request by yannickchristhomas**

### CODE REVIEW

File: src/main/groovy/com/cloudogu/gitops/config/ApplicationConfigurator.groovy [hunk 1]
```diff
@@ -51,7 +51,7 @@ class ApplicationConfigurator {
                             chart  : 'docker-registry',
                             repoURL: 'https://charts.helm.sh/stable',
                             version: '1.9.4'
-                    ]
+                    ],
```
Comment (schnatterer) [inline src/main/groovy/com/cloudogu/gitops/config/ApplicationConfigurator.groovy:54]: this change is not necessary

Comment (schnatterer) [inline src/main/groovy/com/cloudogu/gitops/config/ApplicationConfigurator.groovy:54]: this change is not necessary, please undo

File: src/main/groovy/com/cloudogu/gitops/cli/GitopsPlaygroundCli.groovy [hunk 2]
```diff
@@ -186,7 +188,9 @@ class GitopsPlaygroundCli  implements Runnable {
     @Option(names = ['--output-config-file'], description = 'Output current config as config file as much as possible')
     Boolean outputConfigFile
 
-    // args group ArgoCD operator
+    // args group Argo
```
Comment (schnatterer) [inline src/main/groovy/com/cloudogu/gitops/cli/GitopsPlaygroundCli.groovy:191]: this change is not necessary, please undo

File: src/main/groovy/com/cloudogu/gitops/features/argocd/ArgoCD.groovy [hunk 3]
```diff
@@ -335,7 +335,7 @@ class ArgoCD extends Feature {
                             emailToAdmin : config.features['argocd']['emailToAdmin']
                     ],
                     registry : [
-                            twoRegistries: config.registry['twoRegistries']
+                            twoRegistries: config.registry['twoRegistries'],
```
Comment (schnatterer) [inline src/main/groovy/com/cloudogu/gitops/features/argocd/ArgoCD.groovy:338]: this change is not necessary, please undo

File: src/main/groovy/com/cloudogu/gitops/features/Jenkins.groovy [hunk 4]
```diff
@@ -79,6 +79,8 @@ class Jenkins extends Feature {
             globalPropertyManager.setGlobalProperty("${config.application['namePrefixForEnvVars']}REGISTRY_PATH", config.registry['path'] as String)
         }
 
+        globalPropertyManager.setGlobalProperty("${config.application['namePrefixForEnvVars']}MAVEN_CENTRAL_MIRROR", config.jenkins['mavenCentralMirror'] as String)
```
Comment (schnatterer) [inline src/main/groovy/com/cloudogu/gitops/features/Jenkins.groovy:82]: I'd like to have a unit test  `Sets maven mirror` (or similar) for this.

What happens if mavenCentralMirror is not set? I also like to have a unit test for this (you can add those in `'Maps config properly'()` next to the other `verify(globalPropertyManager, never())` checks.

**Review by schnatterer — CHANGES_REQUESTED** [review]
After working through my comments please rebase on main.

After starting the GOP, both `petclinic-plain` and `petclinic-helm` failed with this:
````
java.io.FileNotFoundException: The specified user settings file does not exist: /tmp/gitops-playground-jenkins-agent/workspace/xample-apps_petclinic-plain_main/.m2/settings.xml
````
After restarting them, the builds succeeded. Did those work fine for you?


Please also do a negative test: When `--maven-central-mirror` is not set, we don't want a an empty env var here
http://jenkins.localhost/manage/configure

image


For the record, this is how I tested

```
 docker run --rm -t  -u $(id -u)  \
  -v ~/.config/k3d/kubeconfig-gitops-playground.yaml:/home/.kube/config \
    --net=host \
      gitops-playground:dev --yes -x --argocd --base-url=http://localhost --ingress-nginx  --maven-central-mirror https://maven-central.storage.googleapis.com/maven2/

File: src/test/groovy/com/cloudogu/gitops/features/JenkinsTest.groovy [hunk 5]
```diff
@@ -111,7 +111,7 @@ class JenkinsTest {
         verify(globalPropertyManager, never()).setGlobalProperty(eq('MY_PREFIX_REGISTRY_PULL_PATH'), anyString())
         verify(globalPropertyManager, never()).setGlobalProperty(eq('MY_PREFIX_REGISTRY_PUSH_URL'), anyString())
         verify(globalPropertyManager, never()).setGlobalProperty(eq('MY_PREFIX_REGISTRY_PUSH_PATH'), anyString())
-        
+
```
Comment (schnatterer) [inline src/test/groovy/com/cloudogu/gitops/features/JenkinsTest.groovy:114]: Please add the never check here.
This is where we check the default or "happy" path. That is, no mavenMirror.

**Review by schnatterer — CHANGES_REQUESTED** [review]
Almost there :+1:  :slightly_smiling_face: 
Maybe I confused you with the comment about the test. If so, lets talk about it.

**Review by schnatterer — APPROVED** [review]
